main.rb

```diff
@@ -0,0 +1,173 @@
+require_relative 'person'
+require_relative 'student'
+require_relative 'book'
+require_relative 'classroom'
+require_relative 'teacher'
+require_relative 'rental'
+
+class App
+  def initialize
+    @people = []
+    @books = []
+    @rentals = []
+  end
+
+  def start
+    puts 'Welcome to our School Library!'
+    loop do
+      menu
+      option = gets.chomp
+      break if option == '7'
+
+      get_num option
+    end
+    puts 'Thanks for using our library!'
+  end
+
+  def menu
+    puts
+    puts 'Please choose an option by entering a number: '
+    puts '1 - List all books'
+    puts '2 - List all people'
+    puts '3 - Create a person'
+    puts '4 - Create a book'
+    puts '5 - Create a rental'
+    puts '6 - List all rentals for a given person id'
+    puts '7 - Exit'
+  end
+
+  def get_num(option)
+    case option
+    when '1'
+      list_all_books
+    when '2'
+      list_all_people
+    when '3'
+      create_person
+    when '4'
+      create_book
+    when '5'
+      create_rental
+    when '6'
+      list_rentals_by_person_id
+    else
+      puts 'Please enter a number between 1 and 7.'
+    end
+  end
+
+  def list_all_books
+    puts 'No books in the database! Please add a book.' if @books.empty?
+
+    @books.each { |book| puts "Title: #{book.title}, Author: #{book.author}" }
+    sleep 0.75
+  end
+
+  def list_all_people
+    puts 'No people in the database! Please add a person.' if @people.empty?
+    @people.map { |person| puts "[#{person.class}] Name: #{person.name}, ID: #{person.id}, Age: #{person.age}" }
+    sleep 0.75
+  end
+
+  def create_person
+    print 'Would you like to create a student(1) or a teacher(2) [Please input a number]: '
+    option = gets.chomp
+
+    case option
+    when '1'
+      create_student
+    when '2'
+      create_teacher
+    else
+      puts 'Invalid input. Please input 1 or 2'
+    end
+  end
+
+  def create_student
+    print 'Age: '
+    age = gets.chomp.to_i
+
+    print 'Name: '
+    name = gets.chomp
+
+    print 'Has parent permission? [Y/N]: '
+    parent_permission = gets.chomp.downcase
+
+    student = Student.new(@class, age, name, parent_permission)
+    @people << student
+
+    puts 'Student created successfully'
+    sleep 0.75
+  end
+
+  def create_teacher
+    print 'Age: '
+    age = gets.chomp.to_i
+
+    print 'Name: '
+    name = gets.chomp
+
+    print 'Specialization: '
+    specialization = gets.chomp
+
+    teacher = Teacher.new(age, specialization, name)
```

Comment:
- [ ] Please I am unable to create a teacher because of the error the screenshot below. Please kindly fix that, since it is a requirement for this project, the app has to be running. I think it has to do with the number of parameters passed to the `teacher` instance object. Please check your `teacher class in your teacher.rb file`
![teacher](https://user-images.githubusercontent.com/65852621/152136135-63646570-1f6f-4ed4-82e7-84d9c52ff0f8.png)

Reply:
Issue fixed